### PR TITLE
refactor: Use step length in `VolumeMaterialInteraction`

### PR DESCRIPTION
**Pull request by andiwand**
WIPWIP

fixes https://github.com/acts-project/acts/issues/2279

### CODE REVIEW

File: Core/include/Acts/Propagator/detail/VolumeMaterialInteraction.hpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ struct VolumeMaterialInteraction {
       slab = MaterialSlab(navigator.currentVolume(state.navigation)
                               ->volumeMaterial()
                               ->material(pos),
-                          1);  // state.stepping.StepSize
+                          state.stepping.stepSize.value());
```
Comment (andiwand) [inline Core/include/Acts/Propagator/detail/VolumeMaterialInteraction.hpp:93]: this should be set to the previous step length

**Comment by andiwand** [comment]
cc @Corentin-Allaire

**Comment by paulgessinger** [comment]
Are we going ahead with this @andiwand?

File: Core/include/Acts/Propagator/detail/VolumeMaterialInteraction.hpp [hunk 2]
```diff
@@ -92,7 +91,7 @@ struct VolumeMaterialInteraction {
       slab = MaterialSlab(navigator.currentVolume(state.navigation)
                               ->volumeMaterial()
                               ->material(pos),
-                          1);  // state.stepping.StepSize
+                          state.stepping.previousStepSize);
```
Comment (andiwand) [inline Core/include/Acts/Propagator/detail/VolumeMaterialInteraction.hpp:94]: I am not sure anymore if `previousStepSize` is the right thing

**Pull request closed without merging**